api/client/client.go

```diff
@@ -111,19 +177,23 @@ type Config struct {
 	// KeepAliveCount specifies the amount of missed keep alives
 	// to wait for before declaring the connection as broken
 	KeepAliveCount int
-	// TLS is the client's TLS config
-	TLS *tls.Config
+
+	// Credentials are a list of credentials to use when attempting to connect
+	// to Auth.
+	Credentials []Credentials
+
+	// NoPingCheck disables the ping check on client initialization. The ping
+	// check is used to verify and authenticate the connection, but in cases
+	// where you don't immediately expect the connection to be usable, use
+	// NoPingCheck to disable this check.
+	NoPingCheck bool
```

Comment:
Wouldn't grpc.Dial fail anyway if the connection wasn't available?

Reply:
No, unfortunately it doesn't fail until you try to make a request. The TLS and Dialer can both be bad, and `grpc.Dial` won't care.

Reply:
On a side note, the way we attempt to dial multiple addresses doesn't work properly with `grpc.Dial` either. It will return an error on the first address even if the second is good. Providing a dialer for each address and providing all of them to `grpc.Dial` does seem to work though. I may have to do a deeper dive into how its logic is laid out.

Reply:
@Joerger The reason it doesn't fail is because by default grpc.Dial is non-blocking and establishes connection in background. If you add https://pkg.go.dev/google.golang.org/grpc#WithBlock dial option, it will start failing in case of unhealthy server. I guess our Ping call kinda serves as an analogue of that, though I'm not sure if it can potentially have any race conditions (e.g. we call Ping too soon and connection in background hasn't established yet - not sure what will happen).

Reply:
That explains a lot, thank you. I'll switch out the ping for now.